transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImpl.kt

```diff
@@ -210,19 +210,22 @@ internal class MessagingClientImpl(
                     stateMachine.onError(code, message)
                 }
             }
-            is ErrorCode.WebsocketError -> handleWebSocketError()
+            is ErrorCode.WebsocketError -> handleWebSocketError(
+                ErrorCode.WebsocketError,
+                ErrorMessage.FailedToReconnect
+            )
             else -> log.w { "Unhandled ErrorCode: $code with optional message: $message" }
         }
     }
 
-    private fun handleWebSocketError() {
+    private fun handleWebSocketError(errorCode: ErrorCode, message: String? = null) {
         if (stateMachine.isClosed()) return
         invalidateConversationCache()
         if (reconnectionHandler.shouldReconnect) {
```

Comment:
If the intention is to immediately notify UI that the network has been disabled, we should check to see if the `errorCode` is `NetworkDisabled` and doing something appropriate before running through the current `if … else …` logic. As it currently stands in the event of the socket reporting `NetworkDisabled` it will still rapidly run through the reconnectionHandler until maxAttempts is reached and then end up in the final error state.

Maybe we could add a `invalidate` method to the ReconnectionHandler so if we get network disabled error we can "turn off" the reconnection handler before check to see if to reconnect? e.g.: 

```
if (error == NetworkDisabled)  { reconnectionHandler.invalidate() }

if (reconnectionHandler.shouldReconnect) {
    stateMachine.onReconnect()
    reconnectionHandler.reconnect { connect(true) }
} else {
    stateMachine.onError(errorCode, message)
    attachmentHandler.clearAll()
}
```

Reply:
Good idea. I think I need to refactor this part a little bit.